### PR TITLE
Feat/user object mutation id inputs

### PR DESCRIPTION
**Pull request by justlevine**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This PR enables users to supply either a database ID or a global ID to user mutation inputs of type ID. Part of #998 

Does this close any currently open issues?
------------------------------------------
#2194 

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** Ubuntu 20.04 (wsl2 + devilbox + php8.0.15)

**WordPress Version:** 5.9.2

### CODE REVIEW

File: src/Mutation/UserDelete.php [hunk 1]
```diff
@@ -80,19 +81,21 @@ public static function get_output_fields() {
...
-			/**
-			 * Get the ID from the global ID
-			 */
-			$id_parts = Relay::fromGlobalId( $input['id'] );
+			// Get the user ID.
+			$user_id = Utils::get_database_id_from_id( $input['id'] );
 
-			if ( ! current_user_can( 'delete_users', absint( $id_parts['id'] ) ) ) {
+			if ( empty( $user_id ) ) {
+				throw new UserError( 'The user ID passed is invalid', 'wp-graphql' );
```
Comment (jasonbahl) [inline src/Mutation/UserDelete.php]: should be: 

```
throw new UserError( __( 'The user ID passed is invalid', 'wp-graphql' ) );
```

Comment (jasonbahl) [reply src/Mutation/UserDelete.php]: Github won't let me make a suggestion 🤔

Comment (justlevine) [reply src/Mutation/UserDelete.php]: Gone now I believe

Comment (jasonbahl) [reply src/Mutation/UserDelete.php]: Confirmed.

**Comment by justlevine** [comment]
This also closes #2194

**Review by jasonbahl — DISMISSED** [review]

**Review by jasonbahl — APPROVED** [review]

**Comment by jasonbahl** [comment]
@justlevine so you can change your env variable to make multisite tests run or not

`MULTISTE=0` means the tests will run in non-multisite
`MULTISITE=1` means they will run in multisite

I tend to run tests locally outside of docker (faster for me) so I can run them like so: `MULTISITE=true vendor/bin/codecept run wpunit` or `MULTISITE=false vendor/bin/codecept run wpunit`

**Comment by justlevine** [comment]
> From what I can tell, they're running as expected.

I'm not by my computer, but that's the only reason I can think that the reassign test is failing now with a bad reassign, on _all_ the tests (and not just 5.9 MU). Can be confirmed if you a put a `codecept_debug` in the is_multisite() conditional in that test. Not sure _why_ that would be, though

**Comment by justlevine** [comment]
https://github.com/wp-graphql/wp-graphql/blob/33255ada6524870245a9367be6420e6a37149553/src/Mutation/UserDelete.php#L140

**Review by justlevine — COMMENTED** [review]
Should this be in the `is_multisite()` check, or am I guilty of premature optimization?

File: src/Mutation/UserDelete.php [hunk 2]
```diff
@@ -133,12 +133,22 @@ public static function mutate_and_get_payload() {
 			if ( ! function_exists( 'wpmu_delete_user' ) ) {
 				require_once ABSPATH . 'wp-admin/includes/ms.php';
 			}
+			if ( ! function_exists( 'remove_user_from_blog' ) ) {
```
Comment (justlevine) [inline src/Mutation/UserDelete.php:136]: Should this be in the is_multisite() check, or am I guilty of premature optimization?

**Comment by jasonbahl** [comment]
@justlevine I mistyped. instead of `MULTISITE=false vendor/bin/codecept run wpunit` it needs to be `MULTISITE=0 vendor/bin/codecept run wpunit`

I can verify now by placing this inside the `mutate_and_get_payload` function in the `UserDelete.php` file and running the test command. 

` MULTISITE=0 vendor/bin/codecept run tests/wpunit/UserObjectMutationsTest.php:testDeleteUserWithReassign --debug`

CleanShot 2022-05-16 at 16 56 23

` MULTISITE=1 vendor/bin/codecept run tests/wpunit/UserObjectMutationsTest.php:testDeleteUserWithReassign --debug`

CleanShot 2022-05-16 at 16 57 16

File: src/Mutation/UserDelete.php (hunk 2 above)
Comment (jasonbahl) [reply src/Mutation/UserDelete.php:136]: nope, I was moving this already! The test actually didn't like it outside the `is_multisite()` call! So good suggestion 😄

Comment (jasonbahl) [reply src/Mutation/UserDelete.php]: just pushed this change 😄

**Pull request merged**